pkg/nettest/nettest.go

```diff
@@ -63,7 +63,7 @@ func Retryable(err error) bool {
 	// Using the exact same error check used in:
 	// https://github.com/golang/go/blob/a5d61be040ed20b5774bff1b6b578c6d393ab332/src/net/http/serve_test.go#L1417
 	if errStr := err.Error(); (strings.Contains(errStr, "timeout") && strings.Contains(errStr, "TLS handshake")) ||
-		strings.Contains(errStr, "unexpected EOF") {
+		strings.Contains(errStr, "unexpected EOF") || strings.Contains(errStr, "connection reset by peer") || strings.Contains(errStr, "unexpected http response") {
```

Comment:
Nit: How about `503 Service Unavailable` instead of `unexpected http response` (there could be non-temporary http response that we would like to see). Though maybe it's not an issue because if it's a permanent error then it will retry until the timeout configured in CI...

Reply:
That's a really good point, but I more inclined to leave it as it is (that way we won't need to update this if a different 5xx error pops out), and like you mentioned, any non-transitory error will eventually be shown.